File: src/editors/pixelate.ts
import * as Jimp from "jimp"

export default {
    name: "Pixelate",
    description: "Allows you to pixelate a image.",
    icon: "pixelate.png",
    expectsImageData: true,
    apply: async(partBuff: Buffer): Promise<Buffer> => (await Jimp.read(partBuff)).pixelate(10).getBufferAsync("image/png"),
}

File: core/editors/pixelate.go
package editors

import (
	"github.com/disintegration/gift"
	"github.com/magiccap/MagicCap/assets/editor"
	"image"
)

func init() {
	Editors["pixelate"] = &Editor{
		Name:        "Pixelate",
		Description: "Allows you to pixelate a image.",
		Icon:        editor.Pixelate,
		Apply: func(Region *image.RGBA, _ [3]uint8) *image.RGBA {
			g := gift.New(
				gift.Pixelate(10),
			)
			img := image.NewRGBA(Region.Bounds())
			g.Draw(img, Region)
			return img
		},
	}
}
